### PR TITLE
util: use runtime.Version() to get Go version

### LINKED ISSUE #53618: TiDB fails to build with gotip
## Bug Report

```
$ make GO=gotip server
CGO_ENABLED=1  gotip build -tags codes  -ldflags '-X "github.com/pingcap/tidb/pkg/parser/mysql.TiDBReleaseVersion=v8.2.0-alpha-246-gbdb0545f73" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBBuildTS=2024-05-28 07:11:04" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitHash=bdb0545f73349ac5ce0a96f9397500d3f3e9a58f" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitBranch=master" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBEdition=Community" ' -o bin/tidb-server ./cmd/tidb-server
# github.com/pingcap/tidb/cmd/tidb-server
link: github.com/pingcap/tidb/pkg/util/printer: invalid reference to runtime.buildVersion
make: *** [Makefile:201: server] Error 1
```

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #53618 

Problem Summary:

Related: 

- https://github.com/pingcap/tidb/pull/13763
- https://github.com/golang/go/issues/67401

Alternative solution (not chosen):
```patch
diff --git a/Makefile.common b/Makefile.common
index b540debd42..bab0acd3c9 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -71,6 +71,7 @@ LDFLAGS += -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBBuildTS=$(shell
 LDFLAGS += -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitHash=$(shell git rev-parse HEAD)"
 LDFLAGS += -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitBranch=$(shell git rev-parse --abbrev-ref HEAD)"
 LDFLAGS += -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBEdition=$(TIDB_EDITION)"
+LDFLAGS += "-checklinkname=0"
 
 EXTENSION_FLAG =
 ifeq ($(shell if [ -f pkg/extension/enterprise/.git ]; then echo "true"; fi),true)
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by dveeden** [comment]
```
$ make GO=gotip server && ./bin/tidb-server -V
CGO_ENABLED=1  gotip build -tags codes  -ldflags '-X "github.com/pingcap/tidb/pkg/parser/mysql.TiDBReleaseVersion=v8.2.0-alpha-246-gbdb0545f73-dirty" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBBuildTS=2024-05-28 07:30:49" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitHash=bdb0545f73349ac5ce0a96f9397500d3f3e9a58f" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitBranch=master" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBEdition=Community" ' -o bin/tidb-server ./cmd/tidb-server
Release Version: v8.2.0-alpha-246-gbdb0545f73-dirty
Edition: Community
Git Commit Hash: bdb0545f73349ac5ce0a96f9397500d3f3e9a58f
Git Branch: master
UTC Build Time: 2024-05-28 07:30:49
GoVersion: devel go1.23-37764658 Fri May 24 22:23:55 2024 +0000
Race Enabled: false
Check Table Before Drop: false
Store: unistore
```

File: pkg/util/printer/printer.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"runtime"
 	_ "runtime" // import link package
```
Comment (hawkingrei) [inline pkg/util/printer/printer.go:22]: Can it be deleted?

Comment (hawkingrei) [inline pkg/util/printer/printer.go:22]: Can it be deleted?
```suggestion
	_ "runtime" // import link package
```

**Comment by dveeden** [comment]
This also causes issues for pingcap/tiflow:

```
$ make GO=gotip cdc dm
CGO_ENABLED=0 gotip build  -trimpath  -ldflags '-X "github.com/pingcap/tiflow/pkg/version.ReleaseVersion=v8.2.0-master-dirty" -X "github.com/pingcap/tiflow/pkg/version.BuildTS=2024-05-28 08:12:48" -X "github.com/pingcap/tiflow/pkg/version.GitHash=327b89a0831e53b64e235bcfac3e224cd0b83d10" -X "github.com/pingcap/tiflow/pkg/version.GitBranch=master" -X "github.com/pingcap/tiflow/pkg/version.GoVersion=go version go1.22.3 linux/amd64" -X "github.com/pingcap/tidb/pkg/parser/mysql.TiDBReleaseVersion=v8.2.0-master-dirty"' -o bin/cdc ./cmd/cdc
go: downloading github.com/pingcap/tidb v1.1.0-beta.0.20240520031014-397a460dd067
go: downloading github.com/pingcap/tidb/pkg/parser v0.0.0-20240520031014-397a460dd067
# github.com/pingcap/tiflow/cmd/cdc
link: github.com/pingcap/tidb/pkg/util/printer: invalid reference to runtime.buildVersion
make: *** [Makefile:165: cdc] Error 1
```
So we need to update the dependency once this has been resolved.

**Review by hawkingrei — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**